packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx

```diff
@@ -58,15 +63,15 @@ export const SuiteBanners = () => {
 
     let banner = null;
     let priority = 0;
+    // this handles firmware hash being invalid after a firmware update, not the regular firmware hash check
     if (device?.id && firmwareHashInvalid.includes(device.id)) {
         banner = <FirmwareHashMismatch />;
         priority = 92;
-    } else if (
-        !isFirmwareRevisionCheckDisabled &&
-        isDeviceAcquired(device) &&
-        device.authenticityChecks !== undefined &&
-        device.authenticityChecks.firmwareRevision !== null && // check was performed
-        device.authenticityChecks.firmwareRevision.success === false
```

Comment:
Using `selectFirmwareRevisionCheckError` not only deduplicates logic, but it also fixes a bug :bug: 
This is missing an if for messageSystem `selectIsFeatureDisabled(state, Feature.firmwareRevisionCheck)`

_Currently bug does not present, but if we decided to turn it off globally, this would still be on :scream:  _